### ISSUE TITLE
NoSuchElementException when using container SelenideElement in `container.innerElement.shouldBe(visible, ofSeconds(XX))`

### ISSUE THREAD

**Issue by bereg2k** [selenide/selenide#1455]
## Issue

Here's a code from some Page Object method. Basically, it **selects one of the options in the modal window** using the option name. The modal window is a "container" element for the inner element of the option.

```java
public void selectRecordType(String recordType) {
  var recordTypeSelectContainer = $("div.changeRecordTypeRightColumn");
  recordTypeSelectContainer.$(withText(recordType))
    .shouldBe(visible, ofSeconds(10))
    .click();
}
```

While calling this method in the test **Selenide is throwing `NoSuchElementException` because it's not able to locate `recordTypeSelectContainer` after the default 4 seconds**. It's because the entire modal window appears only after **5-6 seconds**. Using the obligated 10 seconds from `shouldBe()` method, it eventually locates the modal window and the option element inside of it, and it clicks it alright. The test continues with no problem, and it's passed at the end.

Test results will contain this assertion error as follows:

<img width="787" alt="Screenshot 2021-04-21 at 18 04 36" src="https://user-images.githubusercontent.com/42677987/115576708-6ebf7000-a2cc-11eb-9880-6159ab08a50f.png">

Here's what can be extracted from the Allure results:

```
Element not found {div.changeRecordTypeRightColumn}
Expected: exist
Timeout: 4 s.
Caused by: NoSuchElementException: no such element: Unable to locate element: {"method":"css selector","selector":"div.changeRecordTypeRightColumn"}
```

Also, even Selenide's `SimpleReport` contains this "failure", which means that `LogEventListener` is notified about it.

## Expected result
Selenide should take into account the context when locating elements, and, in the case of containers, should only report errors when **the entire chain isn't found in the required timeframe** (e.g. 20 seconds here `container.innerElement.anotherInnerElement.shouldBe(visible, ofSeconds(20))`).

## Additional information
AFAIK, technically, right now Selenide is searching for the root element first, using the default `selenide.timeout`, disregarding the `Duration` value in the terminating `should()` method.

**Comment by asolntsev** [comment]
@bereg2k I have checked: Selenide does use the custom timeout for root element too. Yes, you see "4 s." in the first exception, but it's not a problem because Selenide re-run this search several times until the custom timeout is reached. 

What is really a problem here is that Selenide should not add the first exception to Allure report. It's probably a duplicate of https://github.com/selenide/selenide/issues/1055, but we definitely need to fix both.

**Comment by bereg2k** [comment]
@asolntsev yes, you're right, Selenide retries the search for the container element for a duration set in `should()`. And it throws exceptions on every unsuccessful try – in this case, every 4 seconds that it cannot locate the container. 
We'd have more exceptions if the container appeared after, let's say, 18 seconds, and we used `shouldBe(visible, ofSeconds(20))`.

Regarding #1055. I agree, the root cause seems the same.

**Comment by asolntsev** [comment]
Closing as a duplicate of #1055
